src/notifications/ManagerLinux.cpp

```diff
@@ -79,15 +75,18 @@ NotificationsManager::showNotification(const QString summary,
         static QDBusInterface notifyApp("org.freedesktop.Notifications",
                                         "/org/freedesktop/Notifications",
                                         "org.freedesktop.Notifications");
-        QDBusMessage reply =
-          notifyApp.callWithArgumentList(QDBus::AutoDetect, "Notify", argumentList);
-        if (reply.type() == QDBusMessage::ErrorMessage) {
-                qDebug() << "D-Bus Error:" << reply.errorMessage();
-                return 0;
-        } else {
-                return reply.arguments().first().toUInt();
-        }
-        return true;
+        auto call = notifyApp.asyncCallWithArgumentList("Notify", argumentList);
+        QDBusPendingCallWatcher watcher{QDBusPendingReply{call}};
+        connect(&watcher,
+                &QDBusPendingCallWatcher::finished,
+                this,
+                [&watcher, this, &roomid, &eventid]() {
+                        if (watcher.reply().type() == QDBusMessage::ErrorMessage)
+                                qDebug() << "D-Bus Error:" << watcher.reply().errorMessage();
+                        else
+                                notificationIds[watcher.reply().arguments().first().toUInt()] =
+                                  roomEventId{roomid, eventid};
```

Comment:
Are you sure this code is ever executed? You are connecting to a local variable, so you probably get disconnected immediately again, since the variable gets destroyed. Instead you need to heap allocate it and remove it with deleteLater() in the slot, I think.